api/types/kustomization.go

```diff
@@ -191,6 +191,8 @@ const (
 	deprecatedPatchesStrategicMergeMessage     = "# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead." + " " + deprecatedWarningToRunEditFix
 	deprecatedVarsMessage                      = "# Warning: 'vars' is deprecated. Please use 'replacements' instead." + " " + deprecatedWarningToRunEditFixExperimential
 	deprecatedCommonLabelsWarningMessage       = "# Warning: 'commonLabels' is deprecated. Please use 'labels' instead." + " " + deprecatedWarningToRunEditFix
+	deprecationWarningKind                     = "# Warning: 'kind' field is required in future releases. Please set it explicitly."
+	deprecationWarningAPIVersion               = "# Warning: 'apiVersion' field is required in future releases. Please set it explicitly."
```

Comment:
Since apiVersion and kind should be handled as a pair, I think these two warnings should be combined into a single message like `'apiVersion' and 'kind' field is ...`.